api/src/wfl/executor.clj

```diff
@@ -449,19 +464,36 @@
      executor
      (jdbc/query tx (format query details)))))
 
-(defn ^:private terra-executor-workflows-by-status
-  "Return all the non-retried workflows matching `status` executed by the
-  `executor`."
-  [tx {:keys [details] :as executor} status]
+(defn ^:private remove-nil-and-join
+  "Remove nil elements of `vec` and join with `separator`."
+  ([vec separator]
+   (->> vec (remove nil?) (str/join separator)))
+  ([vec]
+   (remove-nil-and-join vec \space)))
+
+(defn ^:private terra-executor-workflows-by-filters-sql-params
+  "Return sql and params that query `details` for non-retried workflows
+  matching `submission` and/or `status` if specified."
+  [{:keys [details] :as _executor} {:keys [submission status] :as _filters}]
+  (let [optionals (remove-nil-and-join [(when submission "AND submission = ?")
+                                        (when status "AND status = ?")])
+        query     (remove-nil-and-join ["SELECT * FROM %s"
+                                        "WHERE workflow IS NOT NULL"
+                                        "AND retry IS NULL"
+                                        optionals
+                                        "ORDER BY id ASC"])]
+    (->> [submission status]
+         (concat [(format query details)])
+         (remove nil?))))
```

Comment:
And remove the nil inputs from the SQL parameters before querying.

Reply:
Because submission and status are both optional filters, you could have a case where submission is `nil` and status is set, ex. to `"Failed"`.

Without `(remove nil?)` at this stage, we'd ultimately run:

```
(jdbc/query tx
["SELECT * FROM <details> WHERE workflow IS NOT NULL AND retry IS NULL AND status = ? ORDER BY id ASC"
 nil
 "Failed"])
 ```
 
Parameters are then mismatched to placeholders, and parameter list has the wrong cardinality.